components/threading/global_threads/thread_item/attachments/file_card/index.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getAllFiles} from 'mattermost-redux/selectors/entities/files';
+
+import {FileInfo} from 'mattermost-redux/types/files';
+
+import {GlobalState} from 'types/store';
+
+import FileCard from './file_card';
+
+type OwnProps = {
+    id: FileInfo['id'];
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const file = getAllFiles(state)[ownProps.id];
```

Comment:
If you are using functional components, why not use the redux hooks instead?

Reply:
I don't like them to be honest 😅 
I have a couple of reasons too, (no hard selling points against hooks though)

1. It's like "tying" the component to redux, no option to use a component without redux, and just pass a prop.
2. Aesthetics :P, I prefer having less code in a component
3. Composition feels more intuitive, at least to me

That being said, I do use them from time to time :)

Reply:
Unless web platform wants to enforce it (and I haven't heard so far anything like that), your reasons look good to me :smile: